### PR TITLE
Add some doxygen documentation

### PR DESCRIPTION
**Pull request by ToMe25**
This PR adds some doxygen documentation for some methods.

Some things are not documented because i did not know how to/what they do.
Also i am aware the documentation isn't great, but its the best i can reasonably do, while not understanding significant parts of the algorithms used.

I do intend to add some documentation to the other fusion algorithms, but i did not know how to handle their current header.
Please tell me what to do with that so i can add some documentation to them.

For reasons unknown to me doxygen did not work with the default file, so i had to set `INPUT` to `src`.
For this reason i added the Doxyfile to the commit.

I also added the html directory to .gitignore.

### CODE REVIEW

File: src/Adafruit_AHRS_NXPFusion.cpp [hunk 1]
```diff
@@ -302,7 +313,7 @@ void Adafruit_NXPSensorFusion::update(float gx, float gy, float gz, float ax,
     // (with magnitude 1g) NED gravity is along positive z axis
     gSeGyMi[i] = RMi[i][Z];
 
-    // compute a priori acceleration (a-) (g, sensor frame) from decayed a
+    // compute the a priori acceleration (a-) (g, sensor frame) from decayed a
```
Comment (ToMe25) [inline src/Adafruit_AHRS_NXPFusion.cpp:316]: This change was originally part of the changes for my last pr, it seems to have gotten stuck in some cache?

**Comment by ToMe25** [comment]
Why does running doxygen locally tell me i need to document the return type of the NXP begin and update?
Its void and thus shouldn't need any documentation.
It say this:
```
src/Adafruit_AHRS_FusionInterface.h:44: warning: return type of member Adafruit_AHRS_FusionInterface::begin is not documented
src/Adafruit_AHRS_FusionInterface.h:62: warning: return type of member Adafruit_AHRS_FusionInterface::update is not documented
```

**Comment by ladyada** [comment]
oh shoot ok we see whats up. umm can you remove the Doxyfile?

**Comment by ToMe25** [comment]
I can delete it from git(and i just did), but deleting it locally prevents doxygen from running.
We will see whether the CI still works with the file removed tho...

**Comment by ToMe25** [comment]
OK, i deleted it, but now the CI isn't generating anything either, should i just add the Doxyfile back so it works again?

**Comment by ladyada** [comment]
yah thats on us. didnt want to stop your PR :)

**Review by ladyada — APPROVED** [review]

**Pull request merged**